method_comparison/text_generation_benchmark/utils.py

```diff
@@ -331,30 +334,39 @@ def validate_experiment_path(path: str) -> tuple[str, "BenchmarkConfig"]:
 
 
 def get_memory_usage() -> tuple[float, float, float]:
-    """Get current memory usage (RAM and GPU)."""
+    """Get current memory usage (RAM and accelerator)."""
     process = psutil.Process(os.getpid())
     ram_usage_bytes = process.memory_info().rss
     ram_usage_mb = ram_usage_bytes / (1024 * 1024)
 
     if torch.cuda.is_available():
-        gpu_allocated = torch.cuda.memory_allocated()
-        gpu_reserved = torch.cuda.memory_reserved()
-        gpu_allocated_mb = gpu_allocated / (1024 * 1024)
-        gpu_reserved_mb = gpu_reserved / (1024 * 1024)
+        accelerator_allocated = torch.cuda.memory_allocated()
+        accelerator_reserved = torch.cuda.memory_reserved()
+        accelerator_allocated_mb = accelerator_allocated / (1024 * 1024)
+        accelerator_reserved_mb = accelerator_reserved / (1024 * 1024)
+    elif torch.xpu.is_available():
+        accelerator_allocated = torch.xpu.memory_allocated()
+        accelerator_reserved = torch.xpu.memory_reserved()
+        accelerator_allocated_mb = accelerator_allocated / (1024 * 1024)
+        accelerator_reserved_mb = accelerator_reserved / (1024 * 1024)
     else:
-        gpu_allocated_mb = 0.0
-        gpu_reserved_mb = 0.0
+        accelerator_allocated_mb = 0.0
+        accelerator_reserved_mb = 0.0
 
-    return ram_usage_mb, gpu_allocated_mb, gpu_reserved_mb
+    return ram_usage_mb, accelerator_allocated_mb, accelerator_reserved_mb
 
 
-def init_cuda() -> tuple[float, float]:
-    """Initialize CUDA and return initial memory usage."""
+def init_accelerator() -> tuple[float, float]:
+    """Initialize accelerator and return initial memory usage."""
     if torch.cuda.is_available():
         torch.cuda.init()
         torch.cuda.empty_cache()
-        _, gpu_allocated, gpu_reserved = get_memory_usage()
-        return gpu_allocated, gpu_reserved
+        _, accelerator_allocated, accelerator_reserved = get_memory_usage()
+        return accelerator_allocated, accelerator_reserved
+    elif torch.xpu.is_available():
```

Comment:
Here, `elif` is not necessary because the previous branch returns, but it also doesn't hurt.